Lib/tarfile.py

```diff
@@ -2567,7 +2568,19 @@ def chmod(self, tarinfo, targetpath):
         if tarinfo.mode is None:
             return
         try:
-            os.chmod(targetpath, tarinfo.mode)
+            try:
+                os.chmod(targetpath, tarinfo.mode)
+            except OSError as e:
+                if not(hasattr(errno, "EFTYPE") and e.errno == errno.EFTYPE):
+                    raise
+
+                # gh-108948: On FreeBSD, chmod() fails when trying to set the
+                # sticky bit on a file as non-root. But it's a noop in most
+                # other platforms, and we try and match that behavior here.
```

Comment:
Do you mean that on other platforms chmod() silently ignores the sticky bit for non-root users? I'm not sure that I understood correctly.